### ISSUE TITLE
Feature request: an option for YAML output

### ISSUE THREAD

**Issue by who** [springfox/springfox#651]
It'd be handy to have springfox optionally spit out yaml that could be used in [swagger-editor](http://editor.swagger.io/#/).

**Comment by dilipkrish** [comment]
@who would you be interested in submitting a pr for that?

**Comment by who** [comment]
Yes, I'd be willing to have a go at it.

**Comment by dilipkrish** [comment]
Just to be set the expectations. I think these are the things that need to be done
- [ ] Add yaml to the produces in the controllers ```DefaultSwaggerController``` and ```Swagger2Controller```. Currently it defaults to json
- [ ] Add a configuration option to ```@EnableSwagger(supportsYaml=true)``` and ```@EnableSwagger2(supportsYaml=true``` that we use to register a new message converter in the pipeline. 
- [ ] Make sure the swagger modules are registered with the ```ObjectMapper``` in the new message converter  

This is just an initial outline. Obviously you'll find that reality might be a lot different :) Feel free to add to this discussion, just so that we're aware of the direction you're planning to take. 

/c @adrianbk

**Comment by adrianbk** [comment]
Sounds reasonable.

**Comment by rLitto** [comment]
Hi!
How is the development going? Since editor.swagger.io supports yaml, it would be really interesting to have springfox generate it, since we are considering it to document our microservices

**Comment by dilipkrish** [comment]
@rLitto I think @who is prolly working on it. It's easy enough if you want to take a crack at it.

**Comment by rLitto** [comment]
Would like to. Let me check with my chief if I can make it a spare time
story for our company
Il 19/Mag/2015 19:39, "Dilip Krishnan" <notifications@github.com> ha
scritto:

**Comment by who** [comment]
I've sidelined this because of a project priority change.  Also, swagger-editor can import json files now as well, which somewhat fulfills the initial request in the first place.

**Comment by ferrarimarco** [comment]
I needed this and implemented a Spring HttpMessageConverter for application/x-yaml. I'll see if I can produce a pull request for this feature

**Comment by marceloverdijk** [comment]
I would like yaml output as well. Much more readable than json.

**Comment by mjwalfredo** [comment]
This seemed like a fun little project for me to take on in order to get more involved with Springfox so I am taking a crack at implementing this. I think I've got the basics figured out but I was thrown for a bit of a loop when I found that Springfox uses its own ObjectMapper instance in order to keep it isolated from user customizations. I think that I may want to pattern a YamlSerializer class after the JsonSerializer and extract a common interface that will use the appropriate implementation to build the response based on the requested content type.

**Comment by dilipkrish** [comment]
👍 Keep in mind all this _hackery_ to isolate the object mapper might be moot when OAS has its own media type.

**Comment by umbreak** [comment]
It would be pretty useful, since then we could use the output of it directly and automatically in the codegen block for generating the client.

**Comment by dilipkrish** [comment]
@umbreak Yeah the plan is to pull in @apixandru PR

**Comment by beauchef** [comment]
Any news on this? Are you still looking for people to contribute?

**Comment by dilipkrish** [comment]
Absolutely... tho there is already a [PR](https://github.com/springfox/springfox/pull/1488) by @apixandru that I plan on polishing before I pull in

**Comment by sirishankar** [comment]
Can you please let me know when can this be available?

**Comment by orphaner** [comment]
+1

**Comment by sjacek** [comment]
+1

**Comment by julianps** [comment]
+1

**Comment by dilipkrish** [comment]
There is a PR for it. Might revive it.

**Comment by silentsnooc** [comment]
This would be awesome. Apparently [rest-client-generator](https://www.npmjs.com/package/rest-client-generator) accepts URLs already so if there was a `http://localhost:8080/swagger-resources/swagger.yaml` that would be awesome.

**Comment by apixandru** [comment]
Closed the old PR, created a new one here https://github.com/springfox/springfox/pull/2102

**Comment by dilipkrish** [comment]
Thanks @apixandru I'll look to pull this in #2102  shortly

**Comment by Beennnn** [comment]
What a awesome feature ! Can it be merged in the 3-0-0 version ?
Is help still required to merge it ?

**Comment by kanekotic** [comment]
yes, please i would really appreciate a way to export to `yaml`, swagger definitions that are automatically created can also help linting and validating API's, this is a real use case scenario where this bring value vs the current serve only workflow.

**Comment by vijaykumarprabhu** [comment]
Could you please let me know when can this feature be available?

**Comment by ArjunDandagi** [comment]
we are working on a linting tool to consume a yaml file , really interested in this feature

**Comment by ArjunDandagi** [comment]
btw , we are making use of [Zally](https://github.com/zalando/zally)

**Comment by dilipkrish** [comment]
@apixandru has a PR out. Once that is merged it should be available

**Comment by cristina-balan** [comment]
Any news about this feature?
